### PR TITLE
chore: enable lints in firebase_performance

### PR DESCRIPTION
**Pull request by rrousselGit**
## Description

Add extra lints for firebase_performance

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_performance/analysis_options.yaml [hunk 1]
```diff
@@ -1,11 +1,107 @@
-# This is a temporary file to allow us to land a new set of linter rules in a
-# series of manageable patches instead of one gigantic PR. It disables some of
-# the new lints that are already failing on this plugin, for this plugin. It
-# should be deleted and the failing lints addressed as soon as possible.
-
-include: ../../analysis_options.yaml
```
Comment (mikehardy) [inline packages/firebase_performance/analysis_options.yaml:6]: silly question probably, but why (if the all_lint_rules.yaml and analysis_options.yaml are effectively boilerplate duplicated amongst packages) don't they just refer to a common file in firebase_core or similar?

Comment (rrousselGit) [reply packages/firebase_performance/analysis_options.yaml:6]: The goal is, after these PRs are merged, I'll make a separate PR removing all the duplicates that at the root of the repo

Comment (mikehardy) [reply packages/firebase_performance/analysis_options.yaml:6]: makes sense as a pragmatic approach, I was sincerely just curious. Thanks @rrousselGit

**Comment by rrousselGit** [comment]
Merging since apparently the CI fail is not caused by this PR

cc @mikehardy

**Pull request merged**